core/src/main/java/org/apache/iceberg/rest/RESTUtil.java

```diff
@@ -215,8 +232,32 @@ public static String encodeNamespace(Namespace ns) {
    * @return a namespace
    */
   public static Namespace decodeNamespace(String encodedNs) {
-    Preconditions.checkArgument(encodedNs != null, "Invalid namespace: null");
-    String[] levels = Iterables.toArray(NAMESPACE_ESCAPED_SPLITTER.split(encodedNs), String.class);
+    return decodeNamespace(encodedNs, NAMESPACE_ESCAPED_SEPARATOR);
+  }
+
+  /**
+   * Takes in a string representation of a namespace as used for a URL parameter and returns the
+   * corresponding namespace.
+   *
+   * <p>See also {@link #encodeNamespace} for generating correctly formatted URLs.
+   *
+   * @param encodedNamespace a namespace to decode
+   * @param separator The namespace separator to use for decoding. This should be the same separator
```

Comment:
Is this URL-encoded or not the URL-encoded? Allowing both is dangerous, right?

Reply:
we are not enforcing this to be UTF-8 encoded. A user/server could also pass a non-UTF-8 encoded separator

Reply:
There's no validation then and users can specify everything?
No definition & verification of whether this is a single (1 byte) UTF-8 character.
No definition & verification of whether this character is URL-encoded or not.

Reply:
The scope of this PR is to make the existing namespace separator **configurable** and let the server communicate to clients which one should be used. Users can't specify anything as it would always be overriden with what the server sends.